copper-server-impl/src/main/java/com/pogeyan/cmis/impl/services/CmisTypeServices.java

```diff
@@ -1909,22 +1904,37 @@ private static void addIndex(String repositoryId,
 		}
 
 		public static List<? extends TypeDefinition> getTypeDefinitionWithTypePermission(
-				ITypePermissionService typePermissionFlow, String repositoryId, IUserObject role, String typeId) {
+				ITypePermissionService typePermissionFlow, String repositoryId, IUserObject role, String typeId,
+				String tracingId, ISpan parentSpan) {
+			ISpan span = TracingApiServiceFactory.getApiService().startSpan(tracingId, parentSpan,
```

Comment:
we dont need a span in helper APIs, it should only be required as a highlevel flow for tracing.